### PR TITLE
Initial add of Mixerless Telemetry Filter tests

### PR DESCRIPTION
**Pull request by spongeman66**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | Adds tests for Mixerless Telemetry Filters
| License         | Apache 2.0

### What's in this PR?
Adds tests for Mixerless Telemetry Filters

### Why?
Adds system level tests

### Additional context
This PR is not fully working yet. 

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### To Do

- [x] If the PR is not complete but you want to discuss the approach, list what remains to be done here

### CODE REVIEW

File: test/e2e/e2e_helpers_test.go [hunk 1]
```diff
@@ -424,6 +424,82 @@ func sortNamespacedNames(nns []types.NamespacedName) {
...
+//			if k8sapierrors.IsNotFound(err) == expectMissing {
+//				// filter state matches expected
+//				return true, nil
+//			}
+//
+//			//TODO If expectedMissing is true we could be going from F -> Nil or Nil -> F
+//			// Need to make sure the state stays as expected for some minimum time
+//		}
+//	}
+//}
```
Comment (sagikazarmark) [inline test/e2e/e2e_helpers_test.go]:
```suggestion
```

File: test/e2e/e2e_helpers_test.go [hunk 2]
```diff
@@ -424,6 +424,82 @@ func sortNamespacedNames(nns []types.NamespacedName) {
...
+
+func WaitForMixerlessTelemetryFilter(
+	log logr.Logger, namespace, filtername string, expectMissing bool,
+	timeout, interval time.Duration) error {
+
+	return util.WaitForCondition(timeout, interval, func() (bool, error) {
+		_, err := testEnv.Dynamic.Resource(gvr.EnvoyFilter).Namespace(namespace).Get(
+			context.TODO(),
+			filtername,
+			metav1.GetOptions{})
```
Comment (sagikazarmark) [inline test/e2e/e2e_helpers_test.go:480]:
```suggestion
			metav1.GetOptions{},
		)
```

File: test/e2e/e2e_helpers_test.go [hunk 3]
```diff
@@ -424,6 +424,82 @@ func sortNamespacedNames(nns []types.NamespacedName) {
...
 
 func testDataPath(description ginkgo.GinkgoTestDescription) string {
-       path := filepath.Join(description.ComponentTexts...)
-       return strings.ReplaceAll(path, " ", "_")
+	path := filepath.Join(description.ComponentTexts...)
+	return strings.ReplaceAll(path, " ", "_")
 }
+
+func GetIstioObject(istio *istiov1beta1.Istio, namespace, name string) error {
+	return testEnv.Client.Get(context.TODO(), client.ObjectKey{
```
Comment (sagikazarmark) [inline test/e2e/e2e_helpers_test.go:431]: nit: coding style

File: test/e2e/e2e_helpers_test.go [hunk 4]
```diff
@@ -424,6 +424,82 @@ func sortNamespacedNames(nns []types.NamespacedName) {
...
+	log logr.Logger, namespace, filtername string, expectMissing bool,
+	timeout, interval time.Duration) error {
+
+	return util.WaitForCondition(timeout, interval, func() (bool, error) {
+		_, err := testEnv.Dynamic.Resource(gvr.EnvoyFilter).Namespace(namespace).Get(
+			context.TODO(),
+			filtername,
+			metav1.GetOptions{})
+		if err != nil {
+			// log.Error(err, "Get filtername error")
```
Comment (sagikazarmark) [inline test/e2e/e2e_helpers_test.go]: ???

File: test/e2e/e2e_suite_test.go [hunk 5]
```diff
@@ -17,10 +17,15 @@ limitations under the License.
 package e2e
 
 import (
+	"bytes"
+	"encoding/json"
+	"github.com/kr/pretty"
 	"testing"
 	"time"
 
+	_ "encoding/json"
```
Comment (sagikazarmark) [inline test/e2e/e2e_suite_test.go:26]:
```suggestion
```

File: test/e2e/e2e_suite_test.go [hunk 6]
```diff
@@ -17,10 +17,15 @@ limitations under the License.
 package e2e
 
 import (
+	"bytes"
+	"encoding/json"
+	"github.com/kr/pretty"
```
Comment (sagikazarmark) [inline test/e2e/e2e_suite_test.go:22]:
```suggestion
```

File: test/e2e/e2e_suite_test.go [hunk 7]
```diff
@@ -17,10 +17,15 @@ limitations under the License.
...
 	"testing"
 	"time"
 
+	_ "encoding/json"
 	"github.com/go-logr/logr"
+	_ "github.com/kr/pretty"
```
Comment (sagikazarmark) [inline test/e2e/e2e_suite_test.go:28]:
```suggestion
	"github.com/kr/pretty"
```

File: test/e2e/e2e_suite_test.go [hunk 8]
```diff
@@ -80,8 +85,33 @@ var _ = AfterSuite(func() {
...
+		bmap, err = json.Marshal(map[string]ClusterResourceList{"ClusterStateBefore": clusterStateBefore})
+		if err != nil {
+			log.Error(err, "Problem with marshal")
+		} else {
+			err = json.Indent(&prettyJSON, bmap, "", "    ")
+			if err == nil {
+				log.Info("cluster resources before")
+				pretty.Print(string(prettyJSON.Bytes()))
+			}
+		}
```
Comment (sagikazarmark) [inline test/e2e/e2e_suite_test.go:102]:
```suggestion
		if err != nil {
			log.Error(err, "Problem with marshal")
			
			return
		}

		err = json.Indent(&prettyJSON, bmap, "", "    ")
		if err == nil {
			log.Info("cluster resources before")
			pretty.Print(string(prettyJSON.Bytes()))
		}
```

Similar changes are recommended below.

File: test/e2e/e2e_test.go [hunk 9]
```diff
@@ -17,7 +17,11 @@ limitations under the License.
 package e2e
 
 import (
+	"context"
 	"fmt"
+	"github.com/banzaicloud/istio-operator/pkg/util"
+	"strings"
+	//	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
```
Comment (sagikazarmark) [inline test/e2e/e2e_test.go:23]:
```suggestion
```

File: test/e2e/e2e_test.go [hunk 10]
```diff
@@ -17,7 +17,11 @@ limitations under the License.
 package e2e
 
 import (
+	"context"
 	"fmt"
+	"github.com/banzaicloud/istio-operator/pkg/util"
```
Comment (sagikazarmark) [inline test/e2e/e2e_test.go]: Should move this down to the group below.

**Comment by spongeman66** [comment]
a bad email was added to git, so unable to allow CLA.
This PR has been replaced by
https://github.com/banzaicloud/istio-operator/pull/663

**Pull request closed without merging**